core/nullDC.cpp

```diff
@@ -40,35 +40,41 @@ settings_t settings;
 #include <windows.h>
 #endif
 
-int GetFile(char *szFileName, char *szParse=0, u32 flags=0)
+int GetFile(std::string& diskFileName)
 {
-	cfgLoadStr("config","image",szFileName,"null");
-	if (strcmp(szFileName,"null")==0)
+	cfgLoadStr("config", "image", diskFileName, "");
+#if HOST_OS==OS_WINDOWS
+	if (diskFileName.empty())
 	{
-	#if HOST_OS==OS_WINDOWS
+		diskFileName.resize(MAX_PATH);
 		OPENFILENAME ofn;
-		ZeroMemory( &ofn , sizeof( ofn));
-	ofn.lStructSize = sizeof ( ofn );
-	ofn.hwndOwner = NULL  ;
-	ofn.lpstrFile = szFileName ;
-	ofn.lpstrFile[0] = '\0';
-	ofn.nMaxFile = MAX_PATH;
-	ofn.lpstrFilter = "All\0*.*\0\0";
-	ofn.nFilterIndex =1;
-	ofn.lpstrFileTitle = NULL ;
-	ofn.nMaxFileTitle = 0 ;
-	ofn.lpstrInitialDir=NULL ;
-	ofn.Flags = OFN_PATHMUSTEXIST|OFN_FILEMUSTEXIST ;
-
-		if (GetOpenFileNameA(&ofn))
+		ZeroMemory(&ofn, sizeof(ofn));
+		ofn.lStructSize = sizeof(ofn);
+		ofn.lpstrFile = &diskFileName[0];
+		ofn.nMaxFile = MAX_PATH;
+		ofn.lpstrFilter = "Supported Disk Image Files(.CDI, .GDI)\0*.CDI;*.GDI\0\0";
+		ofn.nFilterIndex = 1;
+		ofn.lpstrTitle = "Open disk image file";
+		ofn.Flags = OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
+
+		while (!GetOpenFileName(&ofn))
 		{
-			//already there
-			//strcpy(szFileName,ofn.lpstrFile);
+			const DWORD errorCode = CommDlgExtendedError();
+			if (errorCode == FNERR_BUFFERTOOSMALL)
+			{
+				//The first two bytes of the lpstrFile buffer contain an integer value specifying the size required to receive the full name, in characters.
+				const size_t newBufferSize = ((size_t)ofn.lpstrFile[0] | (size_t)ofn.lpstrFile[1] << 8);
+				diskFileName.resize(newBufferSize);
+				//try again
+			}
+            else
+            {
+                return (errorCode == ERROR_SUCCESS) ? rv_ok : rv_error;
```

Comment:
I think returning ``rv_error`` here prevents the BIOS from booting if no file is specified.
Under Linux we boot the BIOS if the filename is wrong (the file does not exists) but since we have a file open dialog under Windows we shouldn't do that. However we should boot to BIOS if the user clicks on cancel in the dialog.
I mean, he just started reicast, why should he directly cancel?